specification/appcomplianceautomation/resource-manager/Microsoft.AppComplianceAutomation/preview/2022-11-16-preview/appcomplianceautomation.json

```diff
@@ -0,0 +1,1374 @@
+        "modelAsString": true
+      }
+    },
+    "ControlStatus": {
+      "description": "Indicates the control status.",
+      "enum": [
+        "Passed",
+        "Failed",
+        "NotApplicable"
+      ],
+      "readOnly": true,
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ControlStatus",
+        "modelAsString": true
+      }
+    },
+    "AssessmentSeverity": {
+      "description": "Indicates the assessment severity.",
+      "enum": [
+        "High",
+        "Medium",
+        "Low"
+      ],
+      "readOnly": true,
+      "type": "string",
+      "x-ms-enum": {
+        "name": "AssessmentSeverity",
+        "modelAsString": true
+      }
+    },
+    "ControlType": {
+      "description": "Indicates the control type.",
+      "enum": [
+        "FullyAutomated",
+        "PartiallyAutomated",
+        "Manual"
+      ],
+      "readOnly": true,
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ControlType",
+        "modelAsString": true
+      }
+    },
+    "ControlFamilyType": {
+      "description": "Indicates the control family type.",
+      "enum": [
+        "FullyAutomated",
+        "PartiallyAutomated",
+        "Manual"
+      ],
+      "readOnly": true,
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ControlFamilyType",
+        "modelAsString": true
+      }
+    },
+    "CategoryType": {
+      "description": "Indicates the compliance category type.",
+      "enum": [
+        "FullyAutomated",
+        "PartiallyAutomated",
+        "Manual"
+      ],
+      "readOnly": true,
+      "type": "string",
+      "x-ms-enum": {
+        "name": "CategoryType",
+        "modelAsString": true
+      }
+    },
+    "ResourceStatus": {
+      "description": "Indicates the resource status.",
+      "enum": [
+        "Healthy",
+        "Unhealthy",
+        "NotApplicable"
+      ],
+      "readOnly": true,
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ResourceStatus",
+        "modelAsString": true
+      }
+    },
+    "Assessment": {
+      "type": "object",
+      "description": "A class represent the assessment.",
+      "properties": {
+        "name": {
+          "description": "The name of the assessment.",
+          "type": "string",
+          "readOnly": true
+        },
+        "severity": {
+          "$ref": "#/definitions/AssessmentSeverity",
+          "readOnly": true,
+          "description": "The severity level of this assessment."
+        },
+        "description": {
+          "description": "The description of the assessment.",
+          "type": "string",
+          "readOnly": true
+        },
+        "remediation": {
+          "description": "The remediation of the assessment.",
+          "type": "string",
+          "readOnly": true
+        },
+        "isPass": {
+          "type": "boolean",
+          "readOnly": true,
+          "description": "Indicates whether all the resource(s) are compliant."
```

Comment:
consider updating to enum  [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

Reply:
@raosuhas Do you mean change to like "true", "false" string enum?

Reply:
Hi @raosuhas ~
Fixed, please review [here](https://github.com/Azure/azure-rest-api-specs/pull/21152/files#diff-16cc0066db31d47ee8b4f0e9385ece034881677d74a14b4d48790444b24bda4dL838).